packages/blade/src/components/Input/BaseInput/BaseInputTagSlot.native.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { ScrollView, Text as RNText } from 'react-native';
+import styled from 'styled-components';
+import type { BaseInputTagSlotProps } from './types';
+import { BASEINPUT_DEFAULT_HEIGHT, BASEINPUT_WRAPPER_MAX_HEIGHT } from './baseInputConfig';
+import BaseBox from '~components/Box/BaseBox';
+import { makeSize } from '~utils';
+import { size } from '~tokens/global';
+
+const StyledScrollView = styled(ScrollView)((_props) => {
+  return {
+    display: 'flex',
+    flexDirection: 'column',
+    // gap is still not working in RN for some reason
+    // gap: makeSpace(props.theme.spacing[3]),
+  };
+});
+
+const ScrollableTagSlotContainer = ({
+  maxTagRows,
+  children,
+  handleOnClick,
+}: Pick<BaseInputTagSlotProps, 'maxTagRows' | 'showAllTags' | 'handleOnClick'> & {
+  children: (React.ReactNode | null)[];
+}): React.ReactElement => {
+  const scrollViewRef = React.useRef<ScrollView>(null);
+  const [isScrolling, setIsScrolling] = React.useState(false);
+  return (
+    <StyledScrollView
+      ref={scrollViewRef}
+      contentContainerStyle={{
+        flexWrap: 'wrap',
+        position: 'relative',
+      }}
+      onScrollBeginDrag={() => {
+        setIsScrolling(true);
```

Comment:
We should check if(!isScrolling) setIsScrolling(true). Drag event will get triggered and constantly

Reply:
@saurabhdaware ^

Reply:
It will be the same thing no? As long as the previous and current state is same it shouldn't cause any perf issues right 🤔 

https://pgarciacamou.medium.com/react-doesnt-always-trigger-a-re-render-on-setstate-4644212560a